purchase_requisition_line_description/tests/test_purchase_requisition_line_description.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import common
+from odoo import fields
+
+
+class TestPurchaseRequisitionLineDescription(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestPurchaseRequisitionLineDescription, cls).setUpClass()
+        partner = cls.env['res.partner'].create({
+            'name': 'Test partner',
+        })
+        cls.product = cls.env['product.product'].create({
+            'name': 'Product',
+            'standard_price': 10,
+            'description_purchase': 'description for purchase',
+        })
+        cls.purchase = cls.env['purchase.requisition'].create({
+            'partner_id': partner.id,
+            'line_ids': [(0, 0, {
+                'product_id': cls.product.id,
+                'name': cls.product.name,
+                'price_unit': 79.80,
+                'product_qty': 15.0,
+                'product_uom': cls.env.ref('uom.product_uom_unit').id,
+                'date_planned': fields.Date.today(),
+            })]
+        })
+
+    def test_onchange_product_id(self):
+        self.assertEqual(self.product.name, self.purchase.line_ids[0].name)
+        # Test onchange product
+        self.purchase.line_ids[0]._onchange_product_id()
```

Comment:
IMO would be better to use `Form` (https://github.com/odoo/odoo/blob/12.0/odoo/tests/common.py#L1075)  to ensure the onchange is triggered automatically. :wink: